var/spack/repos/builtin/packages/intel-parallel-studio/package.py

```diff
@@ -305,6 +305,13 @@ def setup_environment(self, spack_env, run_env):
             # MPI. It is also possible to set the variables in the modules.yaml
             # file if Intel MPI is the dominant, or only, MPI on a system.
             run_env.set('I_MPI_ROOT', join_path(self.prefix, 'impi'))
+            # Set mpi environment variables
+            spack_env.set('I_MPI_CC', spack_cc)
```

Comment:
I am not sure but in `openmpi` these things are done in `def setup_dependent_environment(self, spack_env, run_env, dependent_spec):`.  see the docs http://spack.readthedocs.io/en/latest/packaging_guide.html?highlight=setup_dependent_environment
`setup_environment` appears to be mostly modules related http://spack.readthedocs.io/en/latest/module_file_support.html?highlight=setup_environment